sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -195,17 +203,26 @@ public virtual Response<DetectedLanguage> DetectLanguage(string document, string
             try
             {
                 var documents = new List<LanguageInput>() { ConvertToLanguageInput(document, countryHint) };
-                Response<LanguageResult> result = _serviceRestClient.Languages(new LanguageBatchInput(documents), cancellationToken: cancellationToken);
-                Response response = result.GetRawResponse();
+                var input = new LanguageDetectionAnalysisInput();
+                foreach (var doc in documents)
+                {
+                    input.Documents.Add(doc);
+                }
+                var analyzeLanguageDetection = new AnalyzeTextLanguageDetectionInput { AnalysisInput = input };
+                Response<AnalyzeTextTaskResult> result = _cognitiveRestClient.AnalyzeText(analyzeLanguageDetection, cancellationToken: cancellationToken);
 
-                if (result.Value.Errors.Count > 0)
+                var languageDetection = result.Value as LanguageDetectionTaskResult;
```

Comment:
before you were doing a [cast expression](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/type-testing-and-cast#cast-expression) `var languageDetection = (LanguageDetectionTaskResult)result.Value;` and here you are using the `as` [operator](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/type-testing-and-cast#as-operator).

which do you think should be the patter we should use everywhere?